filament/include/filament/Texture.h

```diff
@@ -202,6 +202,17 @@ class UTILS_PUBLIC Texture : public FilamentAPI {
          */
         Builder& swizzle(Swizzle r, Swizzle g, Swizzle b, Swizzle a) noexcept;
 
+        /**
+         * Associate an optional tag with this Texture for debugging purposes.
+         *
+         * name will show in error messages and should be kept as short as possible.
+         * The name string must persist until build() has been called on this Builder.
+         *
+         * @param name A short string to identify this Texture
+         * @param len Length name tag, or 0 to compute the length if name is null-terminated
+         */
+        Builder& name(const char* UTILS_NONNULL name, size_t len = 0) noexcept;
```

Comment:
Also because we will make a copy anyways, there is no benefit in having the restriction that the name must be valid until build() is called. I think we just document that we are making a copy and internally it's just a CString.

With this API in the future we can imagine we do something more efficient that doesn't even need a heap allocation (e.g. we could pass the name in the command stream, and we could copy it into an internal pool).

I think we should restrict the size of `name` to a maximum size. Maybe 128 bytes?